tests/kernel/mutex/sys_mutex/src/main.c

```diff
@@ -402,6 +402,8 @@ ZTEST_USER_OR_NOT(mutex_complex, test_mutex)
 	sys_mutex_unlock(&private_mutex);
 	sys_mutex_unlock(&private_mutex); /* thread_12 should now have lock */
 
+	k_sleep(K_MSEC(1));     /* Give thread_12 a chance to lock */
+
```

Comment:
I may be missing something, but I am not seeing why this is needed on SMP.  The ownership/lock should automatically be assigned to thread_12 before that 2nd sys_mutex_unlock() returns. If there is a race in SMP, then I would have expected it to be due to the pre-existing k_sleep(K_MSEC(1)) on line 400 being too short a time to get thread_12 into the mutex's wait queue.

Reply:
Good point. I have make the sleep on line 400 longer instead.